src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -195,17 +198,81 @@ public int getFee() {
       XdrDataOutputStream xdrOutputStream = new XdrDataOutputStream(outputStream);
-      org.stellar.sdk.xdr.TransactionEnvelope.encode(xdrOutputStream, envelope);
+      TransactionEnvelope.encode(xdrOutputStream, envelope);
       Base64 base64Codec = new Base64();
       return base64Codec.encodeAsString(outputStream.toByteArray());
     } catch (IOException e) {
-      throw new AssertionError(e);
+      throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Decodes a base64-encoded XDR {@link TransactionEnvelope}.
+   *
+   * @bug Does not work on Android, as it performs a call to the method
+   *      decode() of class org.apache.commons.codec.binary.Base64, which
+   *      conflicts with a legacy Apache class included in Android.
+   *      To decode a base64-encoded {@link TransactionEnvelope} on Android,
+   *      use {@code android.util.Base64.decode()}, then pass the result to
+   *      the method {@link #decodeXdrEnvelope(byte[])}.
+   *
+   * @param encodedXdrTxEnvelope the base64-encoding of an XDR-encoded {@link TransactionEnvelope}.
+   * @return the decoded {@link TransactionEnvelope}
+   */
+  public static TransactionEnvelope decodeBase64XdrEnvelope(String encodedXdrTxEnvelope) {
+    checkNotNull(encodedXdrTxEnvelope, "Transaction envelope cannot be null");
+    Base64 base64Codec = new Base64();
+    byte[] xdrTxEnvelope = base64Codec.decode(encodedXdrTxEnvelope);
+    return decodeXdrEnvelope(xdrTxEnvelope);
+  }
+
+  /**
+   * Decodes an XDR {@link TransactionEnvelope}.
+   * @param xdrTxEnvelope the XDR encoded {@link TransactionEnvelope}
+   * @return the decoded {@link TransactionEnvelope}
+   */
+  public static TransactionEnvelope decodeXdrEnvelope(byte[] xdrTxEnvelope) {
+    ByteArrayInputStream inputStream = new ByteArrayInputStream(xdrTxEnvelope);
+    XdrDataInputStream xdrInputStream = new XdrDataInputStream(inputStream);
+    try {
+      return TransactionEnvelope.decode(xdrInputStream);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+  /**
+   * Constructs a Transaction object from a given {@link TransactionEnvelope} object.
+   * @param txEnv
+   * @return the new Transaction object
+   */
+  public static Transaction fromEnvelope(org.stellar.sdk.xdr.TransactionEnvelope txEnv) {
+    checkNotNull(txEnv, "Transaction envelope cannot be null");
+    org.stellar.sdk.xdr.Transaction xdrTx = txEnv.getTx();
+    KeyPair sourceAccount = KeyPair.fromXdrPublicKey(xdrTx.getSourceAccount().getAccountID());
+    long sequenceNumber = xdrTx.getSeqNum().getSequenceNumber().getUint64();
+    org.stellar.sdk.xdr.Operation[] xdrOps = xdrTx.getOperations();
+    Operation[] operations = new Operation[xdrOps.length];
+    for (int i = 0; i < xdrOps.length; i++) {
+      operations[i] = Operation.fromXdr(xdrOps[i]);
+    }
+    Memo memo = Memo.fromXdr(xdrTx.getMemo());
+    org.stellar.sdk.xdr.TimeBounds xdrTimeBounds = xdrTx.getTimeBounds();
```

Comment:
Maybe we can have `fromXdr` on `TimeBounds` too (like `Memo`, `Operation`)?

Reply:
Sure, will do.